### PR TITLE
[SPARK-25533][CORE][WEBUI]AppSummary should hold the information about succeeded Jobs and completed stages only

### PR DESCRIPTION
**Pull request by shahidki31**
## What changes were proposed in this pull request?
Currently, In the spark UI, when there are failed jobs or failed stages, display message for the completed jobs and completed stages are not consistent with the previous versions of spark.
Reason is because, AppSummary holds the information about all the jobs and stages. But, In the below code, it checks against the completedJobs and completedStages. So, AppSummary should hold only successful jobs and stages.

https://github.com/apache/spark/blob/66d29870c09e6050dd846336e596faaa8b0d14ad/core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala#L306 
 https://github.com/apache/spark/blob/66d29870c09e6050dd846336e596faaa8b0d14ad/core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala#L119
So, we should  keep only completed jobs and stage information in the AppSummary, to make it consistent with Spark2.2

## How was this patch tested?
Test steps:
 bin/spark-shell
```
sc.parallelize(1 to 5, 5).collect()
sc.parallelize(1 to 5, 2).map{ x => throw new RuntimeException("Fail")}.collect()
```
**Before fix:**

screenshot from 2018-09-26 03-24-53

screenshot from 2018-09-26 03-25-08

**After fix:**
screenshot from 2018-09-26 03-16-14
screenshot from 2018-09-26 03-16-28

### CODE REVIEW

**Comment by shahidki31** [comment]
Spark 2.2.1 output:
screenshot from 2018-09-26 03-34-56
screenshot from 2018-09-26 03-35-09

**Comment by shahidki31** [comment]
cc @vanzin @srowen

**Comment by vanzin** [comment]
ok to test

File: core/src/main/scala/org/apache/spark/status/AppStatusListener.scala [hunk 1]
```diff
@@ -388,10 +388,11 @@ private[spark] class AppStatusListener(
 
       job.completionTime = if (event.time > 0) Some(new Date(event.time)) else None
       update(job, now, last = true)
+      if(job.status.equals(JobExecutionStatus.SUCCEEDED)) {
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/status/AppStatusListener.scala]: Nits: space after `if`. Can we write `job.stats == JobExecutionStatus.SUCCEEDED`? might be more idiomatic.
Certainly the result of the change looks reasonable, but I don't know this code enough to be sure it's the right change. It seems reasonable.

**Comment by shahidki31** [comment]
Hi @vanzin , The build has passed.

**Comment by vanzin** [comment]
Merging to master / 2.4 / 2.3.

**Comment by vanzin** [comment]
(Small conflict in 2.4 that I'm fixing myself and testing locally.)

**Comment by vanzin** [comment]
There's a separate issue with the test in master / 2.4 that I'll fix before I push your change to 2.4 / 2.3.

**Comment by shahidki31** [comment]
Thank you @vanzin for merging.

**Pull request closed without merging**